packages/dialog/test/dialog.test.js

```diff
@@ -161,4 +162,28 @@ describe('vaadin-dialog', () => {
       expect(contentMinWidth).to.be.gt(0);
     });
   });
+
+  describe('focus restoring', () => {
+    let dialog, button;
+
+    beforeEach(() => {
+      dialog = fixtureSync('<vaadin-dialog></vaadin-dialog>');
+      button = fixtureSync('<button></button>');
+      button.focus();
+    });
+
+    it('should move focus to dialog on open', async () => {
+      dialog.opened = true;
+      await oneEvent(dialog.$.overlay, 'vaadin-overlay-open');
+      expect(isElementFocused(dialog.$.overlay)).to.be.true;
+    });
+
+    it('should restore focus on dialog close', async () => {
+      dialog.opened = true;
+      await oneEvent(dialog.$.overlay, 'vaadin-overlay-open');
+      dialog.opened = false;
+      await aTimeout(0);
+      expect(isElementFocused(button)).to.be.true;
```

Comment:
I have to link a really old but crucial comment by Rolf: https://github.com/vaadin/vaadin-dialog/issues/98#issuecomment-610303588 - this should also make sure that the focus ring is applied ;)

Reply:
Good point, thanks! Haven't considered it.

Reply:
Not sure if you wanna/have to tackle it in this PR, because there is also a separate ticket about https://github.com/vaadin/web-components/issues/437 (also targeting 24.1) - just wanted to mention it, because they are highly related.

Reply:
> this should also make sure that the focus ring is applied ;)

Here is a PR to fix that: https://github.com/vaadin/web-components/pull/5837

> Not sure if you wanna/have to tackle it in this PR, because there is also a separate ticket about https://github.com/vaadin/web-components/issues/437 (also targeting 24.1) - just wanted to mention it, because they are highly related.

Thanks for mentioning. This is an issue that should be resolved separately.